gnoi/cert/client.go

```diff
@@ -37,35 +38,44 @@ func NewClient(c *grpc.ClientConn) *Client {
 }
 
 // Rotate rotates a certificate.
-func (c *Client) Rotate(ctx context.Context, certID string, minKeySize uint32, params pkix.Name, ipAddress string, sign func(*x509.CertificateRequest) (*x509.Certificate, error), caBundle []*x509.Certificate, validate func() error) error {
+func (c *Client) Rotate(ctx context.Context, certID string, minKeySize uint32, params pkix.Name, ipAddress string, sign func(*x509.CertificateRequest) (*x509.Certificate, error), caBundle []*x509.Certificate, validate func() error, debugMessages bool) error {
```

Comment:
This function is >100 lines and as I understand it, it implements three well defined operations in it (Generate, Load, Finalize). For the sake of readability, testability etc.. is there any way you could split this and have three separate functions all called in here? This is just my opinion honestly, I'll it to you guys and @samribeiro to decide :)